src/nsrt_learning/strips_learning/gen_to_spec_learner.py

```diff
@@ -283,6 +287,13 @@ def _try_specializing_pnad(
         new_pnad = self._create_new_pnad_with_params_and_add_effects(
             pnad, updated_params, updated_add_effects)
 
+        if check_datastore_change:
+            # If the new PNAD has a datastore size that's not the same
+            # as that of the original PNAD, then we've potentially lost some
+            # data by specializing, which might do harm!
+            if len(new_pnad.datastore) != len(pnad.datastore):
+                return None
+
```

Comment:
have you verified this PR across several seeds of rnt and rnt single option?

Reply:
good call - just verified results on both envs across 5 seeds! Things work as expected.